packages/@react-aria/test-utils/src/menu.ts

```diff
@@ -108,25 +135,31 @@ export class MenuTester {
     let menu = this.getMenu();
     if (menu) {
       if (!option && optionText) {
-        option = within(menu).getByText(optionText);
+        option = within(menu).getByText(optionText).closest('[role=menuitem], [role=menuitemradio], [role=menuitemcheckbox]');
       }
 
       if (this._interactionType === 'keyboard') {
         if (document.activeElement !== menu || !menu.contains(document.activeElement)) {
           act(() => menu.focus());
         }
 
-        await this.user.keyboard(optionText);
-        await this.user.keyboard('[Enter]');
+        if (option) {
+          await this.keyboardNavigateToOption({option});
+        } else {
+          // valid?
+          await this.user.keyboard(optionText);
```

Comment:
yep this is using typeahead, but lemme know if this was a question about something else

Reply:
ah, valid as in, is this the default way we want to navigate to an option
what if users pass in two which have the almost the same textValue? what if the content is different than the textValue?

Reply:
ah I see, good point. I think that we should get rid of this typeahead interaction then for now since it doesn't feel accurate enough due to the reasons you've listed. If the user wants to simulate typeahead we can think about adding a separate option perhaps, but we can wait for follow up there